# Final Project

# Temporal Data Analysis of the Disney datasets

@author: Ankita Patel

## Foreword

In this notebook, I will be showing Temporal Data Analysis performed on the `Disney` datasets that is located [Here]( https://data.world/kgarrett/disney-character-success-00-16). I am analyzing `Disney` dataset to gain insights into the yearly trends about the `Disney` franchise comparing the gross total and inflated adjusted gross total that helps in making vise decisions about investing into `Disney`.

## Introduction

## Question(s) of interests

As a lifelong Disney enthusiast, I'm driven by a keen interest in the Disney franchise's impact. My primary goal is to conduct a Temporal Analysis of genres from an investor's perspective. This analysis involves a high-level yearly comparison of total gross and adjusted gross values, revealing real growth, long-term movie success, cultural influence, and return on investment. We'll then delve deeper into genre-specific analysis to identify promising options for investors, with a personal fondness for Disney's Comedy genre.

Additionally, Disney's subsidiaries significantly contribute to the franchise's profitability. I'm eager to undertake a revenue-focused analysis of these businesses, offering valuable insights for informed investment decisions within Disney. The results of this comprehensive analysis are sure to be intriguing.


## Dataset Description

The Disney dataset is comprised of five tables: `disney_movies_total_gross.csv`, `disney_revenue_1991-2016.csv`, `disney-character.csv`, `disney-director.csv`, `disney-voice-actor.csv`. All the tables are stored in .csv and contain several `Disney` movies information such as movie titles, release date, genre, MPAA rating, total gross, inflated adjusted gross, revenues, characters, villains, directors. Here we are interested in doing yearly gross and revenue analysis so we would consider, `disney_movies_total_gross.csv` and `disney_revenue_1991-2016.csv`

* **disney_movies_total_gross.csv**
    * This file contains information for the movie titles such as release dates, genre, MPAA rating and total gross and inflated adjusted gross for the movie titles.
* **disney_revenue_1991-2016.csv**
    * This file contains information about the revenue generated by disney subsidiaries and there total revenue for particular year.


## Methods and Results

Since I am mostly concerned by the total gross and Inflated adjusted gross, I will first start by loading in disney_movies_total_gross.csv

Lets start by importing necessary packages required for analysis.

In [1]:
# Lets import all the required libraries needed for this analysis
import pandas as pd
import numpy as np
import altair as alt

# import all the required files (.csv)
movie_gross = pd.read_csv('disney_movies_total_gross.csv', parse_dates = ['release_date'])
disney_revenue = pd.read_csv('disney_revenue_1991-2016.csv')

Lets see what the tables look like.

In [2]:
movie_gross.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

    total_gross inflation_adjusted_gross  
0  $184,925,485           $5,228,953,251  
1   $84,300,000           $2,188,229,052  
2   $83,320,000           $2,187,090,808  
3   $65,000,000           $1,078,510,579  
4   $85,000,000             $920,608,730

Lets get some other information about the **movie_gross** table.

In [3]:
movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   movie_title               579 non-null    object        
 1   release_date              579 non-null    datetime64[ns]
 2   genre                     562 non-null    object        
 3   MPAA_rating               523 non-null    object        
 4   total_gross               579 non-null    object        
 5   inflation_adjusted_gross  579 non-null    object        
dtypes: datetime64[ns](1), object(5)
memory usage: 27.3+ KB


The movie_gross table has 579 rows and 5 columns, every movie_title has release_date, genre, MPAA_rating, total_gross, inflation_adjusted_gross. We could see that the genre contains 562 non-null values instead of 579 and MPAA_rating has 523 non-null values instead of 579. So considering this factor we need to cleanse and manipulate the columns appropriately to analyze the Data more accurately.

Lets get some insights on disney_revenue table.

In [4]:
disney_revenue.head()

Year  Studio Entertainment[NI 1]  Disney Consumer Products[NI 2]  \
0  1991                      2593.0                           724.0   
1  1992                      3115.0                          1081.0   
2  1993                      3673.4                          1415.1   
3  1994                      4793.0                          1798.2   
4  1995                      6001.5                          2150.0   

   Disney Interactive[NI 3][Rev 1]  Walt Disney Parks and Resorts  \
0                              NaN                         2794.0   
1                              NaN                         3306.0   
2                              NaN                         3440.7   
3                              NaN                         3463.6   
4                              NaN                         3959.8   

  Disney Media Networks  Total  
0                   NaN   6111  
1                   NaN   7502  
2                   NaN   8529  
3                   359  10414  
4                   414  12525

Lets gather details on the dtypes and non-null counts for the disney_revenue data frame.

In [5]:
disney_revenue.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26 entries, 0 to 25
Data columns (total 7 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Year                             26 non-null     int64  
 1   Studio Entertainment[NI 1]       25 non-null     float64
 2   Disney Consumer Products[NI 2]   24 non-null     float64
 3   Disney Interactive[NI 3][Rev 1]  12 non-null     float64
 4   Walt Disney Parks and Resorts    26 non-null     float64
 5   Disney Media Networks            23 non-null     object 
 6   Total                            26 non-null     int64  
dtypes: float64(4), int64(2), object(1)
memory usage: 1.5+ KB


The Disney revenue table has 26 rows and 6 columns. The Disney Interactive only has 12 non-null values, limiting precise analysis. Disney Media Networks contains 23 non-null values. To prepare for analysis, we must cleanse the data.

Our focus is on movie release dates and insights from "total_gross" and "inflation_adjusted_gross" columns. We need to convert these columns from object types to int64 for accuracy.

To start cleansing, we remove the dollar sign and commas in the data. A Python function streamlines this process, adhering to the DRY principle for efficiency.

In [7]:
#importing cleansing function from python_function file
import python_function_New as pf
import pandas as pd

# convert the total_gross from object to int type
clean_movie_gross = pf.currency_to_int(movie_gross, 'total_gross')

# convert the inflation_adjusted_gross from object to int type
clean_movie_gross = pf.currency_to_int(clean_movie_gross, 'inflation_adjusted_gross')

print(clean_movie_gross.head())

                       movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross  inflation_adjusted_gross  
0    184925485                5228953251  
1     84300000                2188229052  
2     83320000                2187090808  
3     65000000                1078510579  
4     85000000                 920608730  


In [8]:
clean_movie_gross.dtypes

movie_title                         object
release_date                datetime64[ns]
genre                               object
MPAA_rating                         object
total_gross                          int64
inflation_adjusted_gross             int64
dtype: object

In [9]:
clean_movie_gross['release_year'] = clean_movie_gross['release_date'].dt.year
clean_movie_gross.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross  inflation_adjusted_gross  release_year  
0    184925485                5228953251          1937  
1     84300000                2188229052          1940  
2     83320000                2187090808          1940  
3     65000000                1078510579          1946  
4     85000000                 920608730          1950

Here I have performed chaining method through groupby and aggregating to underdstand the Yearly trends for the total_gross and inflation_adjusted_gross.

In [10]:
yearly_analysis  = clean_movie_gross.groupby('release_year').agg({'movie_title': 'count',
                                                                 'total_gross': 'sum',
                                                                 'inflation_adjusted_gross': 'sum'})
yearly_analysis = yearly_analysis.reset_index()
yearly_analysis

# Rename columns for clarity
yearly_analysis.rename(
    columns={
        'movie_title': 'number_of_movies',
        'total_gross': 'total_gross_sum',
        'inflation_adjusted_gross': 'inflation_adjusted_gross_sum',
    },
    inplace=True,
)
yearly_analysis

release_year  number_of_movies  total_gross_sum  \
0           1937                 1        184925485   
1           1940                 2        167620000   
2           1946                 1         65000000   
3           1950                 1         85000000   
4           1954                 1         28200000   
5           1955                 1         93600000   
6           1959                 1          9464608   
7           1961                 3        188599723   
8           1962                 1          9230769   
9           1963                 1         22182353   
10          1967                 1        141843000   
11          1968                 1         21540050   
12          1970                 2         74282749   
13          1971                 1         17871174   
14          1975                 1         31916500   
15          1977                 4        102717599   
16          1979                 1         35841901   
17          1980                 2         13900000   
18          1981                 4         59899231   
19          1982                 3         38617984   
20          1983                 4         38833732   
21          1984                 2         70999495   
22          1985                 6         72981626   
23          1986                 7        254523258   
24          1987                10        560208813   
25          1988                12        479129298   
26          1989                11        525447310   
27          1990                15        613857056   
28          1991                16        647050340   
29          1992                22        854534354   
30          1993                27        701652109   
31          1994                30       1092255030   
32          1995                32       1131964294   
33          1996                28       1134044721   
34          1997                23        803617066   
35          1998                22       1229279167   
36          1999                21       1200292143   
37          2000                19       1098929717   
38          2001                14        881246784   
39          2002                22       1159269061   
40          2003                19       1564114393   
41          2004                19       1157899879   
42          2005                17        970646842   
43          2006                19       1427356974   
44          2007                14       1436787754   
45          2008                13        968276109   
46          2009                16       1215142753   
47          2010                14       1518975880   
48          2011                14       1103207068   
49          2012                10       1452972057   
50          2013                11       1821352070   
51          2014                12       1514179473   
52          2015                11       2495662696   
53          2016                14       2873393105   

    inflation_adjusted_gross_sum  
0                     5228953251  
1                     4375319860  
2                     1078510579  
3                      920608730  
4                      528279994  
5                     1236035515  
6                       21505832  
7                     1797806719  
8                      109581646  
9                      153870834  
10                     789612346  
11                     138612686  
12                     356362241  
13                      91305448  
14                     131246872  
15                     363659174  
16                     120377374  
17                      43560212  
18                     181636869  
19                     110731153  
20                     100549534  
21                     178132661  
22                     173124212  
23                     577472408  
24                    1186649969  
25                     986841867  
26                    1097851241 

Let's visualize the total_gross_sum and inflation_adjusted_gross_sum using Altair and obtain the mark line graphs to compare the years when Disney movies had high and low impact interms of growth, longitivity, culturally and return on investment.

In [11]:
import altair as alt


chart = alt.Chart(yearly_analysis).mark_line().encode(
    x='release_year:O',
    y='total_gross_sum:Q',
    color=alt.value('blue'),
).properties(
    title='Yearly Total Gross'
) + alt.Chart(yearly_analysis).mark_line().encode(
    x='release_year:O',
    y='inflation_adjusted_gross_sum:Q',
    color=alt.value('red'),
).properties(
    title='Yearly Inflation-Adjusted Gross'
)

chart

alt.LayerChart(...)

Below is the analysis based on the above data visualization:

### Inflation-Adjusted Gross vs. Total Gross Analysis:

#### 1. Real Growth:
Significant spikes in "Inflation-Adjusted Gross" occurred in 1937, 1955, and 1961, suggesting substantial real growth in movie revenue when accounting for inflation.
The years between 1986 and 2007 also showed sustained real growth.

#### 2. Long-Term Success:
The early days (1937-1955) and the period from 1986 to 2007 demonstrated strong long-term financial success.
However, there's no consistent evidence of long-term success after 1961.

#### 3. Cultural Impact:
Disney movies from 1937 to 1955 and between 1986 and 2008 had a significant and lasting cultural impact.
Data after 2008 doesn't provide clear insights into cultural impact.

#### 4. Investment Value:
In 1959 and during 1979-1985, Disney movies resulted in detrimental losses for investors.
After 2008, the growth rate of inflation-adjusted gross declined, possibly indicating reduced investment value.

Now considering the above analysis let's try to check which genere had the highest inflation adjusted gross to understand their impact and to make investment as an investor in Disney franchise.

But before we start conducting this analysis lets gather info for Genre column such as its dtype and if there are any null (NaN) values.

In [12]:
# Examine the rows with null values
movie_gross[movie_gross['genre'].isnull()]

movie_title release_date genre MPAA_rating  \
20   The Many Adventures of Winnie the Pooh   1977-03-11   NaN         NaN   
22               Herbie Goes to Monte Carlo   1977-06-24   NaN         NaN   
23                           The Black Hole   1979-12-21   NaN         NaN   
24                         Midnight Madness   1980-02-08   NaN         NaN   
25            The Last Flight of Noah’s Ark   1980-06-25   NaN         NaN   
26                 The Devil and Max Devlin   1981-01-01   NaN         NaN   
121                                 Newsies   1992-04-08   NaN          PG   
122                             Passed Away   1992-04-24   NaN       PG-13   
128            A Gun in Betty Lou's Handbag   1992-08-21   NaN       PG-13   
146                          Bound by Honor   1993-04-16   NaN           R   
155                     My Boyfriend's Back   1993-08-06   NaN       PG-13   
156                             Father Hood   1993-08-27   NaN       PG-13   
168                           Red Rock West   1994-01-28   NaN           R   
251                         The War at Home   1996-11-20   NaN           R   
304                               Endurance   1999-05-14   NaN          PG   
350                High Heels and Low Lifes   2001-10-26   NaN           R   
355                     Frank McKlusky C.I.   2002-01-01   NaN         NaN   

     total_gross inflation_adjusted_gross  
20            $0                       $0  
22   $28,000,000             $105,847,527  
23   $35,841,901             $120,377,374  
24    $2,900,000               $9,088,096  
25   $11,000,000              $34,472,116  
26   $16,000,000              $48,517,980  
121   $2,706,352               $5,497,481  
122   $4,030,793               $8,187,848  
128   $3,591,460               $7,295,423  
146   $4,496,583               $9,156,084  
155   $3,218,882               $6,554,384  
156   $3,268,203               $6,654,819  
168   $2,502,551               $5,170,709  
251      $34,368                  $65,543  
304     $229,128                 $380,218  
350     $226,792                 $337,782  
355           $0                       $0

We are interested in grouping data based on genre. So, we particularly examine the rows with NaN values in the genre column. Out of the 579 rows in the table, 17 rows have NaN value in the genre column. As our aim is to perform a high level analsis, we can safely remove the rows with null (NaN) values for the genre column.

In [13]:
# Remove the null values from genre column
clean_genre = movie_gross.dropna(subset=['genre'])
clean_genre.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

    total_gross inflation_adjusted_gross  
0  $184,925,485           $5,228,953,251  
1   $84,300,000           $2,188,229,052  
2   $83,320,000           $2,187,090,808  
3   $65,000,000           $1,078,510,579  
4   $85,000,000             $920,608,730

Similarly as we cleaned the dollar ($) and commas (,) from the total_gross and inflation_adjusted_gross columns previously we would need to repeat the similar steps to perform analysis with genre.

In [15]:
import python_function_New as pf
import pandas as pd

# convert the total_gross from object to int type
clean_genre_df = pf.currency_to_int(clean_genre, 'total_gross')

# convert the inflation_adjusted_gross from object to int type
clean_genre_df = pf.currency_to_int(clean_genre_df, 'inflation_adjusted_gross')

print(clean_genre_df.head())

                       movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

   total_gross  inflation_adjusted_gross  
0    184925485                5228953251  
1     84300000                2188229052  
2     83320000                2187090808  
3     65000000                1078510579  
4     85000000                 920608730  


Let's now perform chaining using groupby on Genre column and aggregate movie_title by count, total_gross and inflation_adjusted_gross by sum to gather accurate trend for Genre.

In [16]:
genre_analysis  = clean_genre_df.groupby('genre').agg({'movie_title': 'count',
                                                                 'total_gross': 'sum',
                                                                 'inflation_adjusted_gross': 'sum'})

genre_analysis = genre_analysis.reset_index()
genre_analysis

genre  movie_title  total_gross  inflation_adjusted_gross
0                Action           40   4184563282                5498936786
1             Adventure          129  16389069453               24561266158
2          Black Comedy            3     97543212                 156730475
3                Comedy          182   8119619678               15409526913
4   Concert/Performance            2    103456466                 114821678
5           Documentary           16    180685619                 203488418
6                 Drama          114   4106972970                8195804484
7                Horror            6     87068872                 140483092
8               Musical           16   1157284155                9657565776
9       Romantic Comedy           23   1152206855                1788872933
10    Thriller/Suspense           24   1406806519                2151690954
11              Western            7    359011459                 516709946

Here we are visualizing the total_gorss versus inflation_adjusted_gorss which has grouped considering the genre.

In [17]:
import altair as alt


chart_genre = alt.Chart(genre_analysis).mark_line().encode(
    x='genre:O',
    y='total_gross:Q',
    color=alt.value('blue'),
).properties(
    title='Genre-Total Gross vs. Inflation Adjusted Gross'
) + alt.Chart(genre_analysis).mark_line().encode(
    x='genre:O',
    y='inflation_adjusted_gross:Q',
    color=alt.value('red'),
)

chart_genre

alt.LayerChart(...)

### Genre vs. Movie Performance Analysis:

This analysis explores the performance of movies across various genres based on "Total Gross" and "Inflation-Adjusted Gross." Key insights include:

•	Adventure and Drama genres dominate in terms of both total and inflation-adjusted gross, with a substantial number of movies and high revenue figures.

•	Action and Comedy genres also show strong performance in terms of total gross.

•	Musical genre stands out for having a high inflation-adjusted gross, indicating that these movies maintain their value over time.

•	Romantic Comedy and Thriller/Suspense genres also perform well in generating inflation-adjusted gross.

•	Black Comedy and Horror genres have a lower number of movies and relatively lower revenue figures.

•	Concert/Performance, Documentary, and Western genres have limited representation and revenue in the dataset.

Let us start cleansing and manuplating the disney_revenue Data Frame

In [18]:
# Remove the "Disney Interactive[NI 3][Rev 1]" column
disney_revenue = disney_revenue.drop('Disney Interactive[NI 3][Rev 1]', axis=1)

In [19]:
disney_revenue['Studio Entertainment[NI 1]'].fillna(0, inplace=True)
disney_revenue['Disney Consumer Products[NI 2]'].fillna(0, inplace=True)
disney_revenue['Disney Media Networks'].fillna(0, inplace=True)
disney_revenue

Year  Studio Entertainment[NI 1]  Disney Consumer Products[NI 2]  \
0   1991                      2593.0                           724.0   
1   1992                      3115.0                          1081.0   
2   1993                      3673.4                          1415.1   
3   1994                      4793.0                          1798.2   
4   1995                      6001.5                          2150.0   
5   1996                         0.0                             0.0   
6   1997                      6981.0                          3782.0   
7   1998                      6849.0                          3193.0   
8   1999                      6548.0                          3030.0   
9   2000                      5994.0                          2602.0   
10  2001                      7004.0                          2590.0   
11  2002                      6465.0                          2440.0   
12  2003                      7364.0                          2344.0   
13  2004                      8713.0                          2511.0   
14  2005                      7587.0                          2127.0   
15  2006                      7529.0                          2193.0   
16  2007                      7491.0                          2347.0   
17  2008                      7348.0                          2415.0   
18  2009                      6136.0                          2425.0   
19  2010                      6701.0                          2678.0   
20  2011                      6351.0                          3049.0   
21  2012                      5825.0                          3252.0   
22  2013                      5979.0                          3555.0   
23  2014                      7278.0                          3985.0   
24  2015                      7366.0                          4499.0   
25  2016                      9441.0                             0.0   

    Walt Disney Parks and Resorts Disney Media Networks  Total  
0                          2794.0                     0   6111  
1                          3306.0                     0   7502  
2                          3440.7                     0   8529  
3                          3463.6                   359  10414  
4                          3959.8                   414  12525  
5                          4502.0                 4,142  18739  
6                          5014.0                  6522  22473  
7                          5532.0                  7142  22976  
8                          6106.0                  7512  23402  
9                          6803.0                  9615  25402  
10                         6009.0                  9569  25790  
11                         6691.0                  9733  25360  
12                         6412.0                 10941  27061  
13                         7750.0                 11778  30752  
14                         9023.0                 13207  31944  
15                         9925.0                 14368  34285  
16                        10626.0                 15046  35510  
17                        11504.0                 15857  37843  
18                        10667.0                 16209  36149  
19                        10761.0                 17162  38063  
20                        11797.0                 18714  40893  
21                        12920.0                 19436  42278  
22                        14087.0                 20356  45041  
23                        15099.0                 21152  48813  
24                        16162.0                 23264  52465  
25                        16974.0                 23689  55632

Renaming the Columns for better readability

In [20]:
# Rename columns for clarity
disney_revenue.rename(
    columns={
        'Studio Entertainment[NI 1]': 'Studio Entertainment',
        'Disney Consumer Products[NI 2]': 'Disney Consumer Products',
    },
    inplace=True,
)
disney_revenue

Year  Studio Entertainment  Disney Consumer Products  \
0   1991                2593.0                     724.0   
1   1992                3115.0                    1081.0   
2   1993                3673.4                    1415.1   
3   1994                4793.0                    1798.2   
4   1995                6001.5                    2150.0   
5   1996                   0.0                       0.0   
6   1997                6981.0                    3782.0   
7   1998                6849.0                    3193.0   
8   1999                6548.0                    3030.0   
9   2000                5994.0                    2602.0   
10  2001                7004.0                    2590.0   
11  2002                6465.0                    2440.0   
12  2003                7364.0                    2344.0   
13  2004                8713.0                    2511.0   
14  2005                7587.0                    2127.0   
15  2006                7529.0                    2193.0   
16  2007                7491.0                    2347.0   
17  2008                7348.0                    2415.0   
18  2009                6136.0                    2425.0   
19  2010                6701.0                    2678.0   
20  2011                6351.0                    3049.0   
21  2012                5825.0                    3252.0   
22  2013                5979.0                    3555.0   
23  2014                7278.0                    3985.0   
24  2015                7366.0                    4499.0   
25  2016                9441.0                       0.0   

    Walt Disney Parks and Resorts Disney Media Networks  Total  
0                          2794.0                     0   6111  
1                          3306.0                     0   7502  
2                          3440.7                     0   8529  
3                          3463.6                   359  10414  
4                          3959.8                   414  12525  
5                          4502.0                 4,142  18739  
6                          5014.0                  6522  22473  
7                          5532.0                  7142  22976  
8                          6106.0                  7512  23402  
9                          6803.0                  9615  25402  
10                         6009.0                  9569  25790  
11                         6691.0                  9733  25360  
12                         6412.0                 10941  27061  
13                         7750.0                 11778  30752  
14                         9023.0                 13207  31944  
15                         9925.0                 14368  34285  
16                        10626.0                 15046  35510  
17                        11504.0                 15857  37843  
18                        10667.0                 16209  36149  
19                        10761.0                 17162  38063  
20                        11797.0                 18714  40893  
21                        12920.0                 19436  42278  
22                        14087.0                 20356  45041  
23                        15099.0                 21152  48813  
24                        16162.0                 23264  52465  
25                        16974.0                 23689  55632

Below is the temporal analysis for the percentage in revenue growth, highest total revenue generated in the year, highest revenue growth year, average revenue for each category, 

In [21]:
# 1. Identify the year with the highest total revenue
year_highest_revenue = disney_revenue[disney_revenue['Total'] == disney_revenue['Total'].max()]['Year'].values[0]

# 2. Calculate the percentage change in total revenue from the previous year
disney_revenue['Revenue Change %'] = disney_revenue['Total'].pct_change() * 100

# 3. Find the year with the highest revenue growth
year_highest_growth = disney_revenue[disney_revenue['Revenue Change %'] == disney_revenue['Revenue Change %'].max()]['Year'].values[0]

# 4. Calculate the average revenue for each category
average_revenue = disney_revenue.iloc[:, 1:5].mean()
# 5. Find the category with the highest average revenue
category_highest_avg_revenue = average_revenue.idxmax()


print('\n1. Year with the Highest Total Revenue:', year_highest_revenue)

print('\n2. Percentage Change in Total Revenue from Previous Year:')
print(disney_revenue[['Year', 'Revenue Change %']])
print('\n3. Year with the Highest Revenue Growth:', year_highest_growth)

print('\n4. Average Revenue for Each Category:')
print(average_revenue)

print('\n5. Category with the Highest Average Revenue:', category_highest_avg_revenue)


1. Year with the Highest Total Revenue: 2016

2. Percentage Change in Total Revenue from Previous Year:
    Year  Revenue Change %
0   1991               NaN
1   1992         22.762232
2   1993         13.689683
3   1994         22.101067
4   1995         20.270789
5   1996         49.612774
6   1997         19.926357
7   1998          2.238241
8   1999          1.854109
9   2000          8.546278
10  2001          1.527439
11  2002         -1.667313
12  2003          6.707413
13  2004         13.639555
14  2005          3.876171
15  2006          7.328450
16  2007          3.572991
17  2008          6.569980
18  2009         -4.476389
19  2010          5.294752
20  2011          7.435042
21  2012          3.386888
22  2013          6.535314
23  2014          8.374592
24  2015          7.481614
25  2016          6.036405

3. Year with the Highest Revenue Growth: 1996

4. Average Revenue for Each Category:
Studio Entertainment             6197.150000
Disney Consumer Products         23

/var/folders/yk/h04gr5rj5wlf713dykk_btlh0000gn/T/ipykernel_53209/3317941006.py:11: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  average_revenue = disney_revenue.iloc[:, 1:5].mean()


In [22]:
import altair as alt

total_revenue_chart = alt.Chart(disney_revenue).mark_line().encode(
    x='Year',
    y='Total'
).properties(
    title='Total Revenue Over the Years',
    width=600,
    height=300
)

# Create a bar chart for percentage change in revenue
revenue_change_chart = alt.Chart(disney_revenue).mark_bar().encode(
    x='Year',
    y=alt.Y('Revenue Change %', title='Percentage Change', axis=alt.Axis(format='%')),
    color=alt.condition(
        alt.datum['Revenue Change %'] > 0,
        alt.value('green'),  # Positive change in green
        alt.value('red')    # Negative change in red
    )
).properties(
    title='Percentage Change in Total Revenue from Previous Year',
    width=600,
    height=300
)

# Combine the charts
combined_charts = (total_revenue_chart | revenue_change_chart)

# Show the combined charts
combined_charts

alt.HConcatChart(...)

#### Year with the Highest Total Revenue: 2016

  • The year 2016 had the highest total revenue among the years listed.

#### Percentage Change in Total Revenue from Previous Year:
  
  •	This table shows the percentage change in total revenue from one year to the next. The percentage     change is calculated relative to the previous year. For example, in 1992, the revenue increased       by 22.76% compared to 1991.
  •	The data indicates that revenue generally fluctuated over the years, with both positive and           negative changes.

#### Year with the Highest Revenue Growth: 1996
  
  •	The year 1996 experienced the highest revenue growth in percentage terms. It had a revenue           increase of 49.61% compared to the previous year (1995).

#### Average Revenue for Each Category:
  
  •	The table presents the average revenue for each category. It appears that there are four             categories: "Studio Entertainment," "Disney Consumer Products," "Disney Interactive," and "Walt       Disney Parks and Resorts."
  
  •	"Walt Disney Parks and Resorts" has the highest average revenue among these categories, with an        average of 8512.62 units.
  
  •	"Disney Consumer Products" has the second-highest average revenue, followed by "Studio                Entertainment" and "Disney Interactive."

#### Category with the Highest Average Revenue: Walt Disney Parks and Resorts
  
  •	"Walt Disney Parks and Resorts" is the category with the highest average revenue, making it the        most lucrative category in terms of revenue. This suggests that the theme parks and resorts          business segment is a significant contributor to the company's overall revenue.

Overall, the data provides insights into the company's revenue trends, with 2016 being a standout year for total revenue and Walt Disney Parks and Resorts being the most financially rewarding business category on average.

## Discussions

The temporal data analysis of Disney datasets has revealed significant historical growth in inflation-adjusted gross, with peaks in **1937, 1955, and 1961**. Long-term success was evident in the early years **(1937-1955) and from 1986 to 2007**. The analysis also highlighted **Musical genre** having significant cultural impact in Disney's history. In terms of movie performances based on genres **Adventure and Drama** stood out of the lot, which proved my inclination wrong for **comedy genre**.

The revenue analysis indicated that **2016** was a standout year with the highest total revenue. **"Walt Disney Parks and Resorts"** emerged as the category with the highest average revenue, underlining its significance in contributing to Disney's overall financial performance.
These insights provide valuable guidance for investors interested in the Disney franchise, with the data offering a foundation for informed investment decisions.

Furthermore, it would be interesting to check MPAA ratings impact on the gross total and inflated adjusted gorss total.

## Refrences

Not all the work in this notebook is original. Some parts were borrowed from online resources. I take no credit for parts that are not mine. They were soley used for illustration purposes. Lets give to **Ceasar** what belongs to **Ceasar**.

### Resources Used
* [Data Source](https://data.world/kgarrett/disney-character-success-00-16)
    * This Disney database used in this work was curated by **Kelly Garrett**.
* [Data Cleansing](https://www.kaggle.com/code/tresavarghese/eda-disney-dataset/notebook)
    * Inspiration for generating data cleansing function was understood and taken from **Tresa Varghese**